api/src/api/healthCheckRouter.ts

```diff
@@ -0,0 +1,46 @@
+import type { HealthCheckMetadata, HealthCheckMethod, SuccessfulHealthCheckResponse } from "@domain/types";
+import { Router } from "express";
+
+const requestTimestamp = (): number => Math.round(Date.now() / 1000);
+
+const healthCheckResponseStatus = (response: HealthCheckMetadata): number => {
+  let statusCode = 200;
+
+  if (!response.success && response.error) {
+    statusCode = response.error.timeout ? 504 : 502;
+  }
+
+  return statusCode;
+};
+
+type HealthCheckEndpoint = string;
+type HealthChecks = Map<HealthCheckEndpoint, HealthCheckMethod>;
+
+export const healthCheckRouterFactory = (clients: HealthChecks): Router => {
+  const router = Router();
+
+  router.get("/", (_req, res) => {
+    res.json({
+      timestamp: requestTimestamp(),
+      success: true,
+      data: {
+        message: "Alive",
+      },
+    } as SuccessfulHealthCheckResponse);
```

Comment:
question for my understanding, doesn't this always take the response and transform it into a successful health response even when it is not alive or does that error out in a different way?